### PR TITLE
Run dense layout if trivial layout is not perfect at level 1

### LINKED ISSUE #2845: optimization_level=1 always uses Trivial layout
### Information

- **Qiskit Terra version**: master
- **Python version**:
- **Operating system**:

### What is the current behavior?
```
               ┌───┐                ░ ┌─┐            
q37_0: |0>─────┤ X ├────────────────░─┤M├────────────
          ┌───┐└─┬─┘┌───┐           ░ └╥┘┌─┐         
q37_1: |0>┤ H ├──■──┤ X ├───────────░──╫─┤M├─────────
          ├───┤     └─┬─┘┌───┐      ░  ║ └╥┘┌─┐      
q37_2: |0>┤ H ├───────■──┤ X ├──────░──╫──╫─┤M├──────
          ├───┤          └─┬─┘┌───┐ ░  ║  ║ └╥┘┌─┐   
q37_3: |0>┤ H ├────────────■──┤ X ├─░──╫──╫──╫─┤M├───
          ├───┤               └─┬─┘ ░  ║  ║  ║ └╥┘┌─┐
q37_4: |0>┤ H ├─────────────────■───░──╫──╫──╫──╫─┤M├
          └───┘                     ░  ║  ║  ║  ║ └╥┘
  c5_0: 0 ═════════════════════════════╩══╬══╬══╬══╬═
                                          ║  ║  ║  ║ 
  c5_1: 0 ════════════════════════════════╩══╬══╬══╬═
                                             ║  ║  ║ 
  c5_2: 0 ═══════════════════════════════════╩══╬══╬═
                                                ║  ║ 
  c5_3: 0 ══════════════════════════════════════╩══╬═
                                                   ║ 
  c5_4: 0 ═════════════════════════════════════════╩═
```
```python
backend = provider.get_backend('ibmq_20_tokyo')
new_qc = transpile(qc, backend, optimization_level=1)
```

shows initial layout as the trivial layout, which is disconnected:

Screen Shot 2019-07-22 at 4 14 54 PM

However part #2 in the source says `_layout_check` does better layout 
if the circuit needs swaps, 

```python
# 2. Use a better layout on densely connected qubits, if circuit needs swaps
```
which is obviously the case here, but no other layouts
are actually selected (or imported).

### What is the expected behavior?
Expected that the `DenseLayout` is used if there are swaps needed.

### PR DESCRIPTION
**Pull request by 1ucian0**
This is an alternative for PR #3324

Fixes #2845

Under the assumption that DenseLayout is always better than TrivialLayout (is that true?), this PR is much simpler than using `best of` (PR https://github.com/Qiskit/qiskit-terra/pull/3018). If TriviaLayout does not return a perfect scoring, then it uses DenseLayout.

### CODE REVIEW

File: qiskit/transpiler/preset_passmanagers/level1.py [hunk 1]
```diff
@@ -64,17 +66,18 @@ def level_1_pass_manager(transpile_config):
...
 
     def _choose_layout_condition(property_set):
         return not property_set['layout']
 
-    _choose_layout = TrivialLayout(coupling_map)
-
     # 2. Use a better layout on densely connected qubits, if circuit needs swaps
-    _layout_check = CheckMap(coupling_map)
+    def _not_perfect_yet(property_set):
+        return property_set['trivial_layout_score'] is not None and \
```
Comment (kdk) [inline qiskit/transpiler/preset_passmanagers/level1.py]: `property_set['trivial_layout_score'] is not None` is the guard to make sure we don't write over a user-supplied layout, right?

Comment (1ucian0) [reply qiskit/transpiler/preset_passmanagers/level1.py]: correct.

**Comment by 1ucian0** [comment]
> Can you add a test that when a trivial circuit perfectly maps a circuit, it will not be overwritten (or DenseLayout won't be run)?

Added in ebad9106e

File: qiskit/transpiler/passes/layout/noise_adaptive_layout.py [hunk 2]
```diff
@@ -211,6 +211,11 @@ def run(self, dag):
         for end1, end2, _ in sorted(self.prog_graph.edges(data=True),
                                     key=lambda x: x[2]['weight'], reverse=True):
             self.pending_program_edges.append((end1, end2))
+
+        # In python 3.5, networkx returns edges in different order than in 3.6.
+        # This line can be removed after dropping support for python 3.5
+        self.pending_program_edges.sort()
```
Comment (kdk) [inline qiskit/transpiler/passes/layout/noise_adaptive_layout.py]: Is this redundant with the `sorted` above?

**Pull request merged**